spec.bs

```diff
@@ -632,9 +601,137 @@ To <dfn>obtain a report's shared info</dfn> given an [=aggregatable report=]
 1. Return the result of [=serializing an infra value to a json string=] given
     |sharedInfo|.
 
+Shared Storage API monkeypatches {#shared-storage-api-monkeypatches}
+====================================================================
+
+Issue(43): This should be moved to the Shared Storage spec.
+
+<xmp class="idl">
+partial interface SharedStorageWorkletGlobalScope {
+  readonly attribute PrivateAggregation privateAggregation;
+};
+</xmp>
+
+At the start of each invocation of a Shared Storage operation (i.e. a call to
+`run()` or to `selectURL()`), create a new [=batching scope=] and associate it
+with that invocation.
+
+When a {{PrivateAggregation}} object is exposed on a
+{{SharedStorageWorkletGlobalScope}}, its [=PrivateAggregation/get batching scope
+steps=] should be left default until the `addModule()` execution is complete. At
+that point, it should be set to the following algorithm:
+1. Return the [=batching scope=] associated with the currently executing
+    operation invocation in |scope|.
+
+Note: Multiple operation invocations can be in-progress at the same time, each
+with a different batching scope. However, only one can be currently executing.
+
+When an operation invocation completes, [=mark a batching scope complete=] given
+the associated [=batching scope=], the global scope's [=relevant settings
+object=]'s [=environment settings object/origin=] and
+"<code>shared-storage</code>".
+
+Protected Audience API monkeypatches {#protected-audience-api-monkeypatches}
+============================================================================
+
+Issue(43): This should be moved to the Protected Audience spec, along with any
+    other Protected Audience-specific details.
+
+<xmp class="idl">
+partial interface InterestGroupScriptRunnerGlobalScope {
+  readonly attribute PrivateAggregation privateAggregation;
+};
+
+dictionary PASignalValue {
+  required DOMString baseValue;
+  double scale;
+  (bigint or long) offset;
+};
+
+dictionary PAExtendedHistogramContribution {
+  required (PASignalValue or bigint) bucket;
+  required (PASignalValue or long) value;
+};
+
+[Exposed=InterestGroupScriptRunnerGlobalScope, SecureContext]
+partial interface PrivateAggregation {
+  undefined contributeToHistogramOnEvent(
+      DOMString event, PAExtendedHistogramContribution contribution);
+};
+</xmp>
+
+Issue: Do we want to align naming with implementation?
+
+At the start of each Protected Audience auction (i.e. a call to
+`runAdAuction()`), create a new [=batching scope=] for each unique origin
```

Comment:
Can you state that in terms of an algorithm step? (e.g. before step 1 of https://wicg.github.io/turtledove/#dom-navigator-runadauction)